### ISSUE TITLE
False positive no-member

### ISSUE THREAD

**Issue by maciek-es** [pylint-dev/pylint#10472]
### Bug description

```python
The following code makes pylint report `no-member` for the last line:


from dataclasses import dataclass

@dataclass
class Foo:
    name: str
    val: int

lorem = (Foo(name="lorem", val=0), 0)
ipsum = (Foo(name="ipsum", val=1), 1)
dolor = (Foo(name="dolor", val=2), 2)

names = [sth.name for sth, _ in [lorem, ipsum, dolor]]
```

### Configuration

```ini

```

### Command used

```shell
pylint
```

### Pylint output

```python
pyt.py:14:9: E1101: Instance of 'tuple' has no 'name' member (no-member)
```

### Expected behavior

No error is raised

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.10
Python 3.10.12 (main, May 27 2025, 17:12:29) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22

### Additional dependencies

```python

```

**Comment by Akshay9715** [comment]
The problem appears to originate from **Astroid’s inference of tuple-unpacking inside comprehensions,** not from Pylint itself.
For example:
```python
import astroid
from dataclasses import dataclass

@dataclass
class Foo:
    name: str
    val: int

lorem = (Foo("lorem", 0), 0)
node = astroid.extract_node("[sth.name for sth, _ in [lorem]]")

for inferred in node.elt.infer():
    print(inferred)
```

Output:
```
Uninferable <class 'astroid.util.UninferableBase'>
```

However, the iterable is correctly inferred:
```python
for inferred in node.generators[0].iter.infer():
    print(inferred)
```

Output:
```
List.list(ctx=<Context.Load: 1>, elts=[<Name.lorem ...>])
```

So Astroid can parse [lorem] but fails to propagate tuple-unpacking correctly to sth.
This indicates the bug lies in Astroid’s comprehension or unpacking inference.

**Comment by TheGreatApollyon** [comment]
Root cause found and fixed in astroid: in `for sth, _ in [lorem]`, the name `lorem` (a reference to a sequence) was not inferred before resolving the assignment path into it, so `sth` got the whole tuple instead of its first element, making `sth.name` a `no-member` false positive.

Fixed by https://github.com/pylint-dev/astroid/pull/3182 — names in the iterable are now inferred before the assignment path is resolved into them.
